src/status_im/ui/screens/discover/views.cljs

```diff
@@ -1,95 +1,114 @@
 
-(defn title [label-kw spacing?]
-  [view st/section-spacing
-   [text {:style      (merge (get-in platform-specific [:component-styles :discover :subtitle])
-                             (when spacing? {:margin-top 16}))
-          :uppercase? (get-in platform-specific [:discover :uppercase-subtitles?])
-          :font       :medium}
-    (label label-kw)]])
 
-(defview discover-popular [{:keys [contacts current-account]}]
+(defview top-status-for-popular-hashtag [{:keys [tag current-account]}]
+  (letsubs [discoveries [:get-popular-discoveries 1 [tag]]]
+    [view (merge styles/popular-list-container
+                 (get-in platform/platform-specific [:component-styles :discover :popular]))
+     [view styles/row
+      [view {}
+       [touchable-highlight {:on-press #(do (dispatch [:set :discover-search-tags [tag]])
+                                            (dispatch [:navigate-to :discover-search-results]))}
+        [view {}
+         [text {:style styles/tag-name
+                :font  :medium}
+          (str " #" (name tag))]]]]
+      [view styles/tag-count-container
+       [text {:style styles/tag-count
+              :font  :default}
+        (:total discoveries)]]]
+     [components/discover-list-item {:message         (first (:discoveries discoveries))
+                                     :show-separator? false
+                                     :current-account current-account}]]))
+
+(defview popular-hashtags-preview [{:keys [contacts current-account]}]
   (letsubs [popular-tags [:get-popular-tags 10]]
-    [view st/popular-container
-     [title :t/popular-tags false]
+    [view styles/popular-container
+     [components/title :t/popular-tags false :t/all #(do (dispatch [:set :discover-search-tags (map :name popular-tags)])
+                                                         (dispatch [:navigate-to :discover-all-hashtags]))]
      (if (pos? (count popular-tags))
-       [carousel {:pageStyle st/carousel-page-style
-                  :gap       8
-                  :sneak     16
-                  :count     (count popular-tags)}
+       [carousel/carousel {:pageStyle styles/carousel-page-style
+                           :gap       8
+                           :sneak     16
+                           :count     (count popular-tags)}
         (for [{:keys [name]} popular-tags]
-          [discover-popular-list {:tag             name
-                                  :contacts        contacts
-                                  :current-account current-account}])]
-       [text (label :t/none)])]))
+          [top-status-for-popular-hashtag {:tag             name
+                                           :contacts        contacts
+                                           :current-account current-account}])]
+       [text (i18n/label :t/none)])]))
+
+
+(defn empty-discoveries []
+  [view contacts-st/empty-contact-groups
+   ;; todo change the icon
```

Comment:
;; TODO(nick): The TODO I wrote

Reply:
That's a good point, except that I wasn't the one who originally put this comment in. We can probably dig that out from git, but we'll be changing the icon anyway soon enough.

Reply:
Oh right, ok let's leave it